doc/maintaining/cli.rst

```diff
@@ -0,0 +1,595 @@
+
+If you have trouble running ckan commands, see
+`Troubleshooting ckan Commands`_ below.
+
+.. note::
+
+   Once you activate your CKAN virtualenv the "ckan" command is available from within any 
+   location within the host environment.
+
+   To run a ckan command without activating the virtualenv first, you have
+   to give the full path the ckan script within the virtualenv, for example:
+
+   .. parsed-literal::
+
+      |virtualenv|/bin/ckan -c |/path/to/ckan.ini| user list 
+
+   In the example commands below, we assume you're running the commands with
+   your virtualenv activated and from your ckan directory.
+
+The general form of a CKAN ``ckan`` command is:
+
+.. parsed-literal::
+
+   ckan --config=\ |/path/to/ckan.ini| **command** 
+
+The `` --config`` option tells CKAN where to find your config file, which it
+reads for example to know which database it should use. As you'll see in the
+examples below, this option can be given as ``-c`` for short.
+
+The config file (ckan.ini) will generally be located in the ``/etc/ckan/default/`` directory however it can be located in any directory on the host machine
+
+``command`` should be replaced with the name of the CKAN command that you wish
+to execute. Most commands have their own subcommands and options. 
+
+.. note::
+
+  You may also specify the location of your config file using the CKAN_INI
+  environment variable. You will no longer need to use --config= or -c to
+  tell ckan where the config file is:
+
+  .. parsed-literal::
+
+     export CKAN_INI=\ |/path/to/ckan.ini|
+     
+  You can still run the ckan command in the same directory as the CKAN config file when the config file is named 'development.ini'. You will not be required to use --config or -c in this case. However this usage is deprecated and will be phased out in a future CKAN release.
+  
+  .. parsed-literal::
+
+     |cd /path/to/ckandir;| ckan command
+
+
+.. parsed-literal::
+
+
+Commands and Subcommands
+
+   ckan -c |/path/to/ckan.ini| user list
+
+(Here ``user`` is the name of the CKAN command you're running, and ``list`` is
+a subcommand of ``user``.)
+
+For a list of all available commands, you will need to access `CKAN Commands Reference`_. 
+
+Each command has its own help text, which tells you what subcommands and
+options it has (if any). To print out a command's help text, run the command
+with the ``--help`` option::
+
+   ckan -c |/path/to/ckan.ini| user --help
+
+
+-------------------------------
+Troubleshooting ckan Commands
+-------------------------------
+
+Permission Error
+================
+
+If you receive 'Permission Denied' error, try running ckan with sudo.
```

Comment:
I'm not clear that this is good advice. The postgres commands should be run as 'postgres' user. But none others that I'm aware.

Reply:
This is existing advice. No point digging into this now.